apps/form-gateway/src/gateway/router.ts

```diff
@@ -62,10 +101,29 @@ async function getFile(
   return fileId;
 }
 
-export function findFile(fileApiUrl: URL, tokenProvider: TokenProvider): RequestHandler {
+export function canAccessFile(formResult: SimpleFormResponse, roles: string[], criteria: string) {
+  if (criteria === '{}') return false;
+  if (formResult === null) return false;
+
+  return (
+    formResult?.status === FormStatus.Submitted &&
+    formResult?.submitted !== null &&
+    //Need to check the form gateway roles and form service roles, depending where the request was made.
+    //Whether it was from the form app or directly through the form gateway the roles might be different.
+    roles.find((role) => role.includes(FormServiceRoles.Applicant) || role.includes(ServiceRoles.Applicant))
```

Comment:
Role based is still not specific enough since all applicants will have the applicant role and shouldn't be able to download each other's PDFs. This needs to check the `form.createdBy.id` against the `user.id`. Only the user who created the form should be able to download it. Ideally it would be restricted to the user who submitted the form, but that's not currently captured.